pypeit/spectrographs/spectrograph.py

```diff
@@ -353,39 +409,52 @@ def get_datasec_img(self, filename, det=1, force=True):
         Args:
             filename (str):
                 Name of the file from which to read the image size.
+            section (str):  'datasec' or 'oscansec'
             det (int):
                 Detector number (1-indexed)
-            force (:obj:`bool`, optional):
-                Force the image to be remade
 
         Returns:
             `numpy.ndarray`: Integer array identifying the amplifier
             used to read each pixel.
         """
-        if self.datasec_img is None or force:
-            # Check the detector is defined
-            self._check_detector()
-            # Get the image shape
-            raw_naxis = self.get_raw_image_shape(filename, det=det)
-
-            # This *always* returns spectral then spatial
-            binning = self.get_meta_value(filename, 'binning')
-
-            # This *always* returns spectral then spatial
-            data_sections, one_indexed, include_end \
-                    = self.get_image_section(filename, det, section='datasec')
-
-            # Initialize the image (0 means no amplifier)
-            self.datasec_img = np.zeros(raw_naxis, dtype=int)
-            for i in range(self.detector[det-1]['numamplifiers']):
-                # Convert the data section from a string to a slice
-                datasec = parse.sec2slice(data_sections[i], one_indexed=one_indexed,
-                                          include_end=include_end, require_dim=2,
-                                          binning=binning)
-                # Assign the amplifier
-                self.datasec_img[datasec] = i+1
-
-        return self.datasec_img
+        # Check the detector is defined
+        self._check_detector()
+        # Get the image shape
+        raw_naxis = self.get_raw_image_shape(filename, det=det)
+
+        binning_pypeit = self.get_meta_value(filename, 'binning')
+
+        data_sections, one_indexed, include_end \
+                    = self.get_image_section(filename, det, section=section)
+        # Note on data format
```

Comment:
Put all of this in the docstring so it will be visible on readthedocs (or eventually will be visible once we fix the docs).

Reply:
done